location for "level" parameter

Zatho opened the pull request.

Commit:
location for "level" parameter

vulder commented:
hmm, regarding the model, we should not create a new one with the locations but update the existing one.

ChristianKaltenecker commented:
The alternative group `level` could also be represented by a numeric feature with a numeric range of 1 to 9. 
The representation as a single numeric feature is also better for the feature location since the numeric feature level should only have a single code location (i.e., a single line where the numeric value for level is parsed).

Zatho commented:
<!--
/* Font Definitions */
@font-face
	{font-family:"Cambria Math";
	panose-1:2 4 5 3 5 4 6 3 2 4;}
@font-face
	{font-family:Calibri;
	panose-1:2 15 5 2 2 2 4 3 2 4;}
/* Style Definitions */
p.MsoNormal, li.MsoNormal, div.MsoNormal
	{margin:0cm;
	font-size:11.0pt;
	font-family:"Calibri",sans-serif;}
a:link, span.MsoHyperlink
	{mso-style-priority:99;
	color:blue;
	text-decoration:underline;}
code
	{mso-style-priority:99;
	font-family:"Courier New";}
.MsoChpDefault
	{mso-style-type:export-only;}
@page WordSection1
	{size:612.0pt 792.0pt;
	margin:70.85pt 70.85pt 2.0cm 70.85pt;}
div.WordSection1
	{page:WordSection1;}
-->Yeah, found that very odd in the Model as well.I will compress that feature into one with a range of 1 to 9,if you could send me a quick example what the XML tag Looks like for thatand then I will instead of creating a new file replace the old one as Florian suggested. Von: ChristianKalteneckerGesendet: Donnerstag, 30. September 2021 17:41An: se-sic/ConfigurableSystemsCc: Zatho; AuthorBetreff: Re: [se-sic/ConfigurableSystems] location for "level" parameter (#2) The alternative group level could also be represented by a numeric feature with a numeric range of 1 to 9.The representation as a single numeric feature is also better for the feature location since the numeric feature level should only have a single code location (i.e., a single line where the numeric value for level is parsed).—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.  

ChristianKaltenecker commented:
See [brotli](https://github.com/se-sic/ConfigurableSystems/blob/master/brotli/FeatureModel.xml) for example.
In the tag `numericOptions` are different numeric configuration options with a minumum value, a maximum value and a step function.

Commit:
Updated lrzip FeatureModel.xml

Deleted FeatureModelLocation.xml and added changes to FeatureModel.xml
made compressionLevel a numericOption

Commit:
Updated lrzip FeatureModel.xml

Deleted FeatureModelLocation.xml and added changes to FeatureModel.xml
made compressionLevel a numericOption
Removed optional tag for numericOption

Zatho commented:
<!--
/* Font Definitions */
@font-face
	{font-family:"Cambria Math";
	panose-1:2 4 5 3 5 4 6 3 2 4;}
@font-face
	{font-family:Calibri;
	panose-1:2 15 5 2 2 2 4 3 2 4;}
/* Style Definitions */
p.MsoNormal, li.MsoNormal, div.MsoNormal
	{margin:0cm;
	font-size:11.0pt;
	font-family:"Calibri",sans-serif;}
a:link, span.MsoHyperlink
	{mso-style-priority:99;
	color:blue;
	text-decoration:underline;}
code
	{mso-style-priority:99;
	font-family:"Courier New";}
.MsoChpDefault
	{mso-style-type:export-only;}
@page WordSection1
	{size:612.0pt 792.0pt;
	margin:70.85pt 70.85pt 2.0cm 70.85pt;}
div.WordSection1
	{page:WordSection1;}
-->So ich hab compressionLevel dann mal zu einer numericOption gemacht und das FeatureModel.xml direkt angepasst statt ein extra file zu erstellen.https://github.com/se-sic/ConfigurableSystems/pull/2/commits/580e303e6564065b04a0e5202e991be74425683b  Von: ChristianKalteneckerGesendet: Donnerstag, 30. September 2021 19:16An: se-sic/ConfigurableSystemsCc: Zatho; AuthorBetreff: Re: [se-sic/ConfigurableSystems] location for "level" parameter (#2) See brotli for example.In the tag numericOptions are different numeric configuration options with a minumum value, a maximum value and a step function.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.  

ChristianKaltenecker commented on lrzip/FeatureModel.xml:
```diff
@@ -356,10 +160,25 @@
-    </configurationOption>
-  </binaryOptions>
-  <numericOptions />
+      <optional>False</optional>
```
FYI: numeric options currently can not be declared optional or not (i.e., numeric configuration options are always there). This is a flaw in the SPL Conqueror format.

Other than that, it looks good to me.

Zatho replied:
Alright, I'll throw that out then

Commit:
Merge branch 'lrzip-features' of https://github.com/se-sic/ConfigurableSystems into lrzip-features
